### ISSUE TITLE
Tildes (~) are incorrectly stripped from custom parameter values

### ISSUE THREAD

**Issue by asnare** [divolte/divolte-collector#221]
Custom event parameters with string values containing tildes (~) do not get mapped correctly: the tildes are stripped.

To reproduce:
```javascript
divolte.signal("anEvent", {aParameter: "foo~bar"});
```
and then map:
```groovy
map eventParameters().value('aParameter') onto 'aField'
```
The Avro field has `foobar` with the tilde (~) stripped.

It's unclear where in the chain the stripping occurs: it's present in the URL that hits the server, but not escaped.

The most likely candidate is the HTTP request processing strips it for some reason: in some of the older RFCs tildes weren't allowed in URLs, but they have been allowed for a long time now.

Depending on the cause, other characters and/or fields (e.g. location/referrer) might also be affected.

**Comment by asnare** [comment]
It turns out this bug isn't in the server at all, but in the JavaScript. Specifically the mincoder used for event parameters is supposed to escape tildes (`~`) and exclamation marks (`!`) but this doesn't work correctly.

The impact is that:

 - Tildes get stripped (unless following by another tilde, in which case only the first is stripped.
 - Exclamation marks probably cause a decoding error (unless preceded by a tilde) and all custom parameters are dropped for the event.

**Comment by asnare** [comment]
Fixed via #222.
